crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -38,7 +38,7 @@ pub struct Definition<'db> {
     #[no_eq]
     #[return_ref]
     #[tracked]
-    pub(crate) kind: DefinitionKind<'db>,
+    kind_inner: DefinitionKind<'db>,
```

Comment:
Should the WARNING message above this field now be changed to something like "It's almost always incorrect to access this field diretcly; instead, go via the public `kind()` accessor getter"?

Reply:
I think the fact that it's private should be good enough